github/GithubIntegration.py

```diff
@@ -0,0 +1,140 @@
+        """
+        body = {}
+        if user_id:
+            body = {"user_id": user_id}
+        response = requests.post(
+            f"{self.base_url}/app/installations/{installation_id}/access_tokens",
+            headers={
+                "Authorization": f"Bearer {self.create_jwt()}",
+                "Accept": Consts.mediaTypeIntegrationPreview,
+                "User-Agent": "PyGithub/Python",
+            },
+            json=body,
+        )
+
+        if response.status_code == 201:
+            return github.InstallationAuthorization.InstallationAuthorization(
+                requester=None,  # not required, this is a NonCompletableGithubObject
+                headers={},  # not required, this is a NonCompletableGithubObject
+                attributes=response.json(),
+                completed=True,
+            )
+        elif response.status_code == 403:
+            raise GithubException.BadCredentialsException(
+                status=response.status_code, data=response.text
+            )
+        elif response.status_code == 404:
+            raise GithubException.UnknownObjectException(
+                status=response.status_code, data=response.text
+            )
+        raise GithubException.GithubException(
+            status=response.status_code, data=response.text
+        )
+
+    @deprecated.deprecated("Use get_repo_installation")
+    def get_installation(self, owner, repo):
+        """
+        :calls: `GET /repos/{owner}/{repo}/installation <https://docs.github.com/en/rest/reference/apps#get-a-repository-installation-for-the-authenticated-app>`_
+        :param owner: str
+        :param repo: str
+        :rtype: :class:`github.Installation.Installation`
+        """
+        headers = {
+            "Authorization": f"Bearer {self.create_jwt()}",
+            "Accept": Consts.mediaTypeIntegrationPreview,
+            "User-Agent": "PyGithub/Python",
+        }
+
+        response = requests.get(
+            f"{self.base_url}/repos/{owner}/{repo}/installation",
+            headers=headers,
+        )
+        response_dict = response.json()
+        return github.Installation.Installation(None, headers, response_dict, True)
+
+    def get_repo_installation(self, owner, repo):
+        return self.get_installation(owner, repo)
+
+    def get_installations(self):
+        """
+        :calls: GET /app/installations <https://docs.github.com/en/rest/reference/apps#list-installations-for-the-authenticated-app>
+        :rtype: :class:`github.PaginatedList.PaginatedList[github.Installation.Installation]`
+        """
+        return github.PaginatedList.PaginatedList(
+            contentClass=github.Installation.Installation,
+            requester=github.Requester.Requester(
+                login_or_token=None,
+                password=None,
+                jwt=self.create_jwt(),
+                app_id=None,
+                app_private_key=None,
+                base_url=Consts.DEFAULT_BASE_URL,
```

Comment:
If I'm not mistaken, this line makes the `base_url` passed in through the `__init__` useless. Makes it impossible to use it with Github Enterprise
```suggestion
                base_url=self.base_url,
```

Reply:
Good catch!